### ISSUE TITLE
bug(YouTube/GmsCore support): No connection after the patch

### ISSUE THREAD

**Issue by Joshua64** [inotia00/ReVanced_Extended#2519]
### Type

Other

### Tools used

RVX Manager

### Application

Youtube 19.16.39

### Bug description

After the patch 5.0.1, the patched YouTube app simply does not connect to the internet anymore (home tab, subscription tab, notifications tab, you tab and sometimes even the configuration tab does not open FOR ME)
I patched with everything in default and I used my settings and the default YouTube revanced settings in the app and the patcher but I always have the same error message
And I can not log in because my accounts does not appear 

![Screenshot_2024-12-08-09-40-02-303_app rvx android youtube](https://github.com/user-attachments/assets/782b9086-e669-41c0-824e-46b63cfce32d)

The same app but with previous patch (and yes I have two versions installed because I have two YouTube accounts and one is for one account (my 
personal account) and the other is for the other account (or my secondary account)

![Screenshot_2024-12-08-09-54-01-734_com rvx android youtube](https://github.com/user-attachments/assets/e49a1bf4-d40a-4eb6-8e49-ceac950b1e9e)

Everything works fine without any problem
Now I can not use my second account because do not work :') (and yes I do not have one backup of the previous patch :') )

### Device Environment

Xiaomi Redmi Note 13 4g 8gb Ram
Hyper OS 1.0
Android 14 (with the latest updates and security patches)

![Screenshot_2024-12-08-10-00-24-340_com android settings](https://github.com/user-attachments/assets/fb8b8716-fd04-421a-8623-95052b5e1a94)

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I did not use any settings marked as `Experimental Flags`.
- [X] I have patched the APK according to the [documentation](https://github.com/inotia00/revanced-documentation#readme).
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by Joshua64** [comment]
https://drive.google.com/file/d/1IE0DH4zSEj-pRjb1GW0gP8HJ1ux4R7su/view?usp=sharing

https://drive.google.com/file/d/1Sz-oB3d-2lyB2hw2tPTOD0xpF54nYK_S/view?usp=sharing

I have this two

**Comment by inotia00** [comment]
It doesn't reproduce on my side

It may be an old bug in GmsCore

Remove the account registered to GmsCore in system settings and log in again

If the issue is still not resolved, follow these steps:

1. Backup GmsCore and RVX
2. Uninstall GmsCore and RVX
3. Install GmsCore first
4. Install RVX

**Comment by YT-Advanced** [comment]
This bug seem being reported by many users. Can u temporary revert https://github.com/inotia00/revanced-patches/commit/787dd5990e66fe4b4ef8db9ca7e8841d2ff27fee and https://github.com/inotia00/revanced-patches/commit/96f2db9d3b7c44cb8a0b75437a73303ac3e67150 to check

**Comment by giwrgosmant** [comment]
Same issue here, nothing I tried solved the issue.
With ReVanced and RVX Music GMSCore account working fine tho.

**Comment by ArcherLegend** [comment]
> Same issue here, nothing I tried solved the issue. With ReVanced and RVX Music account in GMSCore detected and working fine tho.

Yeah, I have temporarely patched anddea to "workaround" the problem as he didn't merge the 5.0.1 patches yet.

**Comment by giwrgosmant** [comment]
> > Same issue here, nothing I tried solved the issue. With ReVanced and RVX Music account in GMSCore detected and working fine tho.
> 
> Yeah, I have temporarely patched anddea to "workaround" the problem as he didn't merge the 5.0.1 patches yet.

I even tried to go back to previously patched v4.16.1 and now I get the same error. 🤷 No idea what's going on.

**Comment by ArcherLegend** [comment]
> Same issue here, nothing I tried solved the issue. With ReVanced and RVX Music account in GMSCore detected and working fine tho.

Yeah, I have temporarely patched anddea to "workaround" the problem as he didn't merge the 5.0.1 patches yet.

@Joshua64 can you removed the screenshots from your report and attach them as files? They make navigating the bug report pretty tedious at the moment as they take a lot of screen space.

**Comment by NMBlanco** [comment]
YouTube doesnt even open for me after patching, it just crashes immediately.

**Comment by mysticmac3** [comment]
Can confirm the same issue

**Comment by kjxxxx** [comment]
Confirm. "You're offline" error. Reinstalled GmsCore, reinstalled RVX, re-added account and nothing.

RVX Music works like a charm.

**Comment by rsaurav12** [comment]
Good thing is i backed up my opd patched apk

**Comment by jacobh12349** [comment]
It's happening for me too, I got the error from my logcat too, if that helps.

[rvxerrors.txt](https://github.com/user-attachments/files/18052798/rvxerrors.txt)

**Comment by Shiroe93** [comment]
Same issue here

**Comment by esperanzality** [comment]
me too 
cant connect

**Comment by Avelayy** [comment]
Same issue here

**Comment by esperanzality** [comment]
me too

**Comment by Karthik-Subbu** [comment]
I am still facing please check network connection error while attempting to signil in even with patch version 5.0.2.

**Comment by Avelayy** [comment]
Microg-RE works:  https://github.com/WSTxda/MicroG-RE/releases/tag/5.9

**Comment by ArcherLegend** [comment]
I can confirm that the issue is still present on my end with patches 5.0.2
I am so sorry inotia00 for the hassle you are facing with the 5.0.X release of the patches.

**Comment by inotia00** [comment]
Reopening the issue as it is still a valid issue

**Comment by inotia00** [comment]
The issue is caused by incorrect constants used during the migration process to ReVanced Patcher 21

This issue will be fixed soon

**Comment by inotia00** [comment]
fixed in [revanced-patches-5.0.3](https://github.com/inotia00/revanced-patches/releases/tag/v5.0.3)

**Comment by STEK1337** [comment]
> fixed in [revanced-patches-5.0.3](https://github.com/inotia00/revanced-patches/releases/tag/v5.0.3)

Still not working for me
![Screenshot_20241209_154141_YouTube](https://github.com/user-attachments/assets/5a5897f4-0709-4855-83ee-3674f60bf96e)
, tried removing and adding account with no luck

**Comment by esperanzality** [comment]
me too

**Comment by ArcherLegend** [comment]
For me it is now working for the moment. Try to remove the accounts, uninstall gmscore and RVX and reinstall everything.

I have to specify that it works for now for me, maybe in 1 or 2 hours it will stop working like the last time.

**Comment by inotia00** [comment]
It doesn't seem to be a patch related issue

![5-0-3](https://github.com/user-attachments/assets/17395a58-1922-4ae9-a757-a378d76325ca)

I tested it on Android Emulator, and the detailed process is as follows:

1. Uninstalled both GmsCore and RVX before starting the test
2. Installed GmsCore (non-Huawei) first
3. Patched with revanced-patches-5.0.3
4. Installed the patched RVX on the device
5. Opened RVX and logged in to my Google account
6. Also checked that RVX was registered in GmsCore Cloud Messaging

※ When I did the same process with revanced-patches-5.0.2, RVX can't load my Google account

**Comment by STEK1337** [comment]
Reinstalling microG did the trick, it works now.

**Comment by esperanzality** [comment]
After uninstalling and restarting, I was able to connect. Thank you so much!

**Comment by KainXVIII** [comment]
> Microg-RE works: https://github.com/WSTxda/MicroG-RE/releases/tag/5.9

Yes, replaced mmicrog with this app and yr rex works alright!
